packages/backend-data/src/convert_schema.test.ts

```diff
@@ -93,4 +171,221 @@ void describe('convertSchemaToCDK', () => {
+        )
+        .returns(a.ref('post'))
+        .authorization([a.allow.public()]),
+    });
+
+    const convertedDefinition = convertSchemaToCDK(
+      modified,
+      secretResolver,
+      stableBackendIdentifiers
+    );
+
+    assert.equal(
+      Object.values(convertedDefinition.dataSourceStrategies).length,
+      1
+    );
+    assert.deepEqual(
+      Object.values(convertedDefinition.dataSourceStrategies)[0],
+      {
+        dbType: 'POSTGRES',
+        // eslint-disable-next-line spellcheck/spell-checker
+        name: '00034dcf3444861c3ca5postgresql',
+        dbConnectionConfig: {
+          connectionUriSsmPath:
+            '/amplify/testBackendId/testBranchName-branch-e482a1c36f/POSTGRES_CONNECTION_STRING',
+        },
+        customSqlStatements: {
+          '../test-assets/test-sql-handler/oddList.sql':
+            'SELECT * from post where id % 2 = 1;',
+        },
+        vpcConfiguration: undefined,
+      }
+    );
+  });
+
+  void it('produces expected definition for Postgresql schema with inline custom query reference', () => {
+    const postgresSchema = configure({
+      database: {
+        engine: 'postgresql',
+        connectionUri: new TestBackendSecret('POSTGRES_CONNECTION_STRING'),
+      },
+    }).schema({
+      post: a
+        .model({
+          id: a.integer().required(),
+          title: a.string(),
+        })
+        .identifier(['id'])
+        .authorization([a.allow.public()]),
+    });
+
+    const modified = postgresSchema.addQueries({
+      oddList: a
+        .query()
+        .handler(a.handler.inlineSql('SELECT * from post where id % 2 = 1;'))
+        .returns(a.ref('post'))
+        .authorization([a.allow.public()]),
+    });
+
+    const convertedDefinition = convertSchemaToCDK(
+      modified,
+      secretResolver,
+      stableBackendIdentifiers
+    );
+
+    assert.equal(
+      Object.values(convertedDefinition.dataSourceStrategies).length,
+      1
+    );
+    assert.deepEqual(
+      Object.values(convertedDefinition.dataSourceStrategies)[0],
+      {
+        dbType: 'POSTGRES',
+        // eslint-disable-next-line spellcheck/spell-checker
+        name: '00034dcf3444861c3ca5postgresql',
+        dbConnectionConfig: {
+          connectionUriSsmPath:
+            '/amplify/testBackendId/testBranchName-branch-e482a1c36f/POSTGRES_CONNECTION_STRING',
+        },
+        customSqlStatements: {},
+        vpcConfiguration: undefined,
+      }
+    );
+  });
+
+  void it('produces expected definition for MySQL schema with vpc config', () => {
+    const postgresSchema = configure({
+      database: {
+        engine: 'mysql',
+        connectionUri: new TestBackendSecret('MYSQL_CONNECTION_STRING'),
+        /* eslint-disable spellcheck/spell-checker */
+        vpcConfig: {
+          vpcId: 'vpc-a1aa11a1',
+          securityGroupIds: ['sg-11111a11'],
+          subnetAvailabilityZones: [
+            {
+              subnetId: 'subnet-1aa1aa11',
```

Comment:
I don't _think_ it'll cause a problem to have duplicated subnetIds in your configs, but it wouldn't be my recommendation, since I believe there's a 1AZ per subnet limit. We don't currently validate that, but if it changes in future this is going to break.

Reply:
These are just test values that don't actually get passed in to the Data CDK construct

Reply:
I.e. part of a unit test only, not e2e